[INTERNAL] ESSupport.md: Add a section about reserved variable names (#840)
SAP/ui5-tooling: docs/pages/ESSupport.md

    });
    ```

#### Reserved Variable Names in a Template Literal

> Template Literals with an expression requires to relinquish following variable names:

Reviewer: From the top of my head, I did not know the meaning of `relinquish` 😄 

How about:

```suggestion
Template Literals with an embedded expression must not use the following variable names:
```

Reviewer 2: The reason for this limitation should be explained. We could even suggest to use https://eslint.org/docs/latest/rules/template-curly-spacing with option "always" to mitigate this. Our placeholder replacement does not allow spaces.

Author: Tackled both comments.

    });
    ```

#### Reserved Variable Names in a Template Literal

> While UI5 Tooling performs a build placeholders are replaced with a values offered by the build. For example `${version}` is replaced with the actual version defined in the package.json of the project. Thereore it is required to not use any **Template Literal** where any expression contains variable with folloing names:

Reviewer 2: ```suggestion
While UI5 Tooling performs a build placeholders are replaced with a values offered by the build. For example `${version}` is replaced with the actual version defined in the package.json of the project. Therefore it is required to not use any **Template Literal** where any expression contains variable with following names:
```
